parsl/monitoring/monitoring.py

```diff
@@ -171,7 +171,10 @@ def start(self, run_id: str, run_dir: str) -> int:
         self.logger.debug("Initializing ZMQ Pipes to client")
         self.monitoring_hub_active = True
 
-        comm_q: Queue[Union[Tuple[int, int], str]]
+        # This annotation is incompatible with typeguard 4.x instrumentation
+        # of local variables, because Queue is not a type: it's a method on
```

Comment:
We should be able to use `multiprocessing.queues.Queue` for the type hint.

Reply:
that fixed it partly, but didn't allow use of `[]` subscripts. But I then made a change that changes which type definition is in use depending on whether you are at runtime or inside mypy. which is pretty ugly.

Reply:
It ain't pretty, but it works.